### PR TITLE
vscode: add 'wayland' and 'egl' use flags

### PR DESCRIPTION
**Pull request by AdelKS**
Hello,

There was something that was bothering me for a while and I just decided to do something about it:
- `wayland-on-wayland`: makes vscode run in native wayland under wayland sessions. I added this use flag to remove the fact that we have two desktop entries (one for x11, one for wayland), and it was tripping my DE (KDE) where the pinned desktop entry would only open new windows even though a window is open. Maybe we should have this flag simply named `wayland` ?
- `egl`: vscode is by default capped to 60fps, on my high refresh rate monitor, navigating in it is annoying. Enabling `egl` removes that framrate capping and makes it fit with the monitor's refresh rate. It works on x11 but it doesn't look like it works in native `wayland`.

Feedback is welcome, and if you think this is a good change, I can mirror it in vscodium before we merge.

Adel

### CODE REVIEW

File: app-editors/vscode/vscode-1.85.0-r2.ebuild [hunk 1]
```diff
@@ -104,10 +104,25 @@ src_install() {
...
+	if use wayland-on-wayland; then
+		EXEC_EXTRA_FLAGS+="--ozone-platform-hint=auto "
+	fi
+	if use egl; then
+		EXEC_EXTRA_FLAGS+="--use-gl=egl "
+	fi
+
+	sed "s|@exec_extra_flags@|$EXEC_EXTRA_FLAGS|g" \
+		"${FILESDIR}/vscode-url-handler.desktop" \
+		> "${T}/vscode-url-handler.desktop" || die
```
Comment (arthurzam) [inline app-editors/vscode/vscode-1.85.0-r2.ebuild:118]:
```suggestion
	local EXEC_EXTRA_FLAGS=()
	if use wayland-on-wayland; then
		EXEC_EXTRA_FLAGS+=( "--ozone-platform-hint=auto" )
	fi
	if use egl; then
		EXEC_EXTRA_FLAGS+=( "--use-gl=egl" )
	fi

	sed "s|@exec_extra_flags@|${EXEC_EXTRA_FLAGS[*]}|g" \
		"${FILESDIR}/vscode-url-handler.desktop" \
		> "${T}/vscode-url-handler.desktop" || die
```

File: app-editors/vscode/metadata.xml [hunk 2]
```diff
@@ -13,6 +13,10 @@
 		<email>proxy-maint@gentoo.org</email>
 		<name>Proxy Maintainers</name>
 	</maintainer>
+	<use>
+		<flag name="wayland-on-wayland">Run in wayland mode in wayland desktops (xwayland otherwise)</flag>
```
Comment (arthurzam) [inline app-editors/vscode/metadata.xml:17]: We talked over that in IRC, let's use `wayland` as the name.

File: app-editors/vscode/vscode-1.85.0-r2.ebuild (hunk 1 above)
Comment (AdelKS) [reply app-editors/vscode/vscode-1.85.0-r2.ebuild]: Thanks! I fixed up the commits with this change

File: app-editors/vscode/metadata.xml (hunk 2 above)
Comment (AdelKS) [reply app-editors/vscode/metadata.xml]: Done

**Review by arthurzam — APPROVED** [review]
Small things I noticed which I'll fix on merge.

USE flags in metadata and in IUSE decl should be sorted :)

**Pull request closed without merging**